.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,127 @@
+name: Run Benchmarks
+
+on:
+  schedule:
+    - cron: '0 1 * * *'  # 2 hrs earlier than sycl-nightly.yml
+  workflow_call:
+    inputs:
+      commit_hash:
+        type: string
+        required: false
+      upload_results:
+        type: string # true/false: workflow_dispatch does not support booleans
+        required: true
+      runner:
+        type: string
+        required: true
+      backend:
+        type: string
+        required: true
+      reset_intel_gpu:
+        type: string  # true/false: workflow_dispatch does not support booleans
+        required: true
+        default: true
+
+  workflow_dispatch:
+    inputs:
+      commit_hash:
+        description: Commit hash to build intel/llvm from
+        type: string
+        required: false
+        default: ''
+      upload_results:
+        description: 'Save and upload results'
+        type: choice
+        options:
+          - false
+          - true
+        default: true
+      runner:
+        type: choice
+        options:
+          - '["PVC_PERF"]'
+      backend:
+        description: Backend to use
+        type: choice
+        options:
+          - 'level_zero:gpu'
+        # TODO L0 V2 support
+      reset_intel_gpu:
+        description: Reset Intel GPUs
+        type: choice
+        options:
+          - false
+          - true
+        default: true
+
+permissions: read-all
+
+jobs:
+  build_sycl:
+    name: Build SYCL from PR
+    if: inputs.commit_hash != ''
```

Comment:
techincally passing empty `commit_hash` is allowed (required == false), so maybe it's not the best way of checking if this is a PR run or nigthly...?

Reply:
hmm tbh, I'm not sure if setting `default: ''` will solve the: `!= ''`, buuut If this will be an issue it can be updated later on